### ISSUE TITLE
Swoole: Server freezes with concurrent requests if a request within request points to the same server

### ISSUE THREAD

**Issue by joysonjoy1811** [laravel/octane#961]
### Octane Version

2.5.9

### Laravel Version

11.x

### PHP Version

8.3.11

### What server type are you using?

Swoole

### Server Version

5.1.4

### Description

I encountered random crashes in a production application running on Laravel Octane with Swoole, where requests would hang indefinitely until timing out. Initially, I suspected a memory leak, but further investigation pointed to "timeout" errors in the Laravel logs from HTTP Facade. Only after these timeouts occur, all subsequent requests become unresponsive until I manually restart the daemon.

The issue was traced to a particular request that was making another request to the same server(though targeting a different endpoint). After running benchmarks locally, I was able to reproduce the problem.

### Steps To Reproduce

```
class TestController extends Controller
{
    public function foo(Request $request)
    {
        $response = Http::asForm()->post('http://127.0.0.1:8000/api/test/bar', [
            'foo' => 'bar'
        ]);
        return $response;
    }

    public function bar(Request $request) {
        return $request->all();
    }
}
```

```
Route::get('/test/foo', [TestController::class, 'foo']);
Route::post('/test/bar', [TestController::class, 'bar']);
```

`php artisan octane:start`
`ab -n 120 -c 24 http://127.0.0.1:8000/api/test/foo`

Output:

```
This is ApacheBench, Version 2.3 <$Revision: 1913912 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking 127.0.0.1 (be patient)...apr_pollset_poll: The timeout specified has expired (70007)
Total of 1 requests completed
```

<img width="1079" alt="Screenshot 2024-10-18 at 9 47 54 PM" src="https://github.com/user-attachments/assets/3edc198f-7afd-4bb6-841a-0b57f4b3a7a2">

**Comment by crynobone** [comment]
Hey there, thanks for reporting this issue.

We'll need more info and/or code to debug this further. Can you please create a repository with the command below, commit the code that reproduces the issue *as one separate commit* on the main/master branch and share the repository here?

Please make sure that you have the [latest version of the Laravel installer](https://github.com/laravel/installer) in order to run this command. Please also make sure you have both Git & [the GitHub CLI tool](https://cli.github.com/) properly set up.

    laravel new bug-report --github="--public"

Do not amend and create a separate commit with your custom changes. After you've posted the repository, we'll try to reproduce the issue.

Thanks!

**Comment by joysonjoy1811** [comment]
You can find the repository here - https://github.com/joysonjoy1811/laravel-bug-report-961

**Comment by NathanFreeman** [comment]
@joysonjoy1811 fixed by swoole5.1.5.  see https://github.com/swoole/swoole-src/releases/tag/v5.1.5

**Comment by crisjohn02** [comment]
I have the same problem eventhough the server is not busy at all.

**Comment by joysonjoy1811** [comment]
@NathanFreeman I can still reproduce the issue in versions 5.1.5 and 5.1.6

**Comment by NathanFreeman** [comment]
The reason for this problem is that the program is stuck in a deadlock, with all processes handling the `/test/foo` request and waiting for a response to the `/test/bar` request. However, the issue is that there are no processes available to handle the `/test/bar` request, resulting in a deadlock. In this situation, the only solution is to start another service to handle the `/test/bar` request separately or to use a `dispatch_func` to differentiate all requests, allowing one process to handle the `/test/bar` request while the remaining processes handle the `/test/foo` requests.

see https://github.com/swoole/swoole-src/issues/5274

**Comment by NathanFreeman** [comment]
The occurrence of this problem is caused by dependencies between requests.

**Comment by mcmc4519** [comment]
I have the same issue, server real freezes with concurrent requests: tha'ts about sse and about parsing other site in controller

**Comment by crisjohn02** [comment]
I have found out this problem persists when there's a arbitrary code execution. I have dynamic php scripts that need to be executed and that causes the freeze. The code is evaluated using eval which is not really a good solution. I have researched and this kind of execution completely defeats the purpose of in-memory app which makes sense. So what I did I refactored my code and eliminate the arbitrary code execution and it works fine now. Hopefully this helps.
